eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitiveObjectHashMap.stg

```diff
@@ -1071,6 +1072,16 @@ public class <name>ObjectHashMap\<V> implements Mutable<name>ObjectMap\<V>, Exte
         return null;
     }
 
+    public Optional\<V> detectOptional(Predicate\<? super V> predicate)
+    {
+        return Optional.ofNullable(this.detect(predicate));
```

Comment:
Very interesting point again (just remembered you mentioned this earlier..). 

> Eventually I'd like to reverse the delegation, detect() should delegate to detectOptional() to better differentiate between finding null and not finding anything. 

How do you envision this reverse delegation work? Is detectOptional() going to return null in case of finding null?